policies/focus-without-user-activation.md

```diff
@@ -0,0 +1,57 @@
+focus-without-user-activation Policy
+===========
+
+The `focus-without-user-activation` policy-controlled feature helps control the use of
+automated focus in a main frame or `<iframe>`.
+
+What does that mean?
+------------
+Automatic focus could happen through:
+  * Use of `autofocus` attribute on a form control (e.g. `<input>`),
+  * Use of scripted focus such as `element.focus()` and `window.focus()`.
+
+Automatic focus is potentially
+[problematic](https://github.com/w3c/webappsec-feature-policy/issues/273) since it provides bad
+embedded content with a tool to steal input focus from the top-level. The proposed feature provides
+a means for developers to block the use of automatic focus in nested contents.
+
+Proposed Solution
+------------
+The proposed feature policy can be used to limit the use of automatic focus. Essentially, when the
+policy is disabled in a document, scripted and automatic focus will only work if the focus has been
+initialized through user activation. This essentially means that `autofocus` will be disabled (
+unless a new element is inserted, with `autofocus`, as a result of user gesture). The scripted focus
+will also only work if it has started with user gesture.
+
+Details on "disabling focus"
+------------
+To avoid unnecessary breakage of existing contents, disabling the focus should still let parts of
+the [focus update steps](https://html.spec.whatwg.org/multipage/interaction.html#focus-update-steps)
+run. In a nutshell:
+  * The `activeElement` of the blocked document should be updated. All the focus related events
```

Comment:
I'm not en expert on focus behavior in browsers, but does that mean that `activeElement` would be updated even though it actually would not be active?

Reply:
This might be a bit hand-wavy and I still need to dig through the steps in the spec, but what I am suggesting is that the document inside the `<iframe>` that is blocked shouldn't know about the focus not really working. I wonder if it makes sense for all the steps in focus work (inside the `<iframe>`) but not make it target for keyboard input (focusable area).

Reply:
I'm not sure what would be the best approach. But out of curiosity: Would you want the document not to know about focus being blocked for security reasons? Anyway, PR looks good to me.